java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -702,6 +704,241 @@ public static CompletableFuture<Long> arrlen(
+     * @param key The key of the JSON document.
+     * @return The object length stored at the root of the document. If document root is not an
+     *     object, an error is raised.<br>
+     *     If <code>key</code> doesn't exist, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\": 1.0, \"b\": {\"a\": {\"x\": 1, \"y\": 2}, \"b\": 2.5, \"c\": true}}").get();
+     * var res = Json.objkeys(client, gs("doc"), gs(".")).get();
+     * assert Arrays.equals((Object[]) res, new Object[] { gs("a"), gs("b") }); // the keys of the object matching the path `.`, which has 2 keys: 'a' and 'b'.
+     * }</pre>
+     */
+    public static CompletableFuture<Object[]> objkeys(
+            @NonNull BaseClient client, @NonNull GlideString key) {
+        return executeCommand(client, new GlideString[] {gs(JSON_OBJKEYS), key});
+    }
+
+    /**
+     * Retrieves the key names in the object values at the specified <code>path</code> within the JSON
+     * document stored at <code>key</code>.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @param path The path within the JSON document.
+     * @return
+     *     <ul>
+     *       <li>For JSONPath (<code>path</code> starts with <code>$</code>):<br>
+     *           Returns an <code>Object[][]</code> with each nested array containing key names for
+     *           each matching object for every possible path, indicating the list of object keys for
+     *           each matching object, or <code>null</code> for JSON values matching the path that are
+     *           not an object. If <code>path</code> does not exist, an empty sub-array will be
+     *           returned.
+     *       <li>For legacy path (<code>path</code> doesn't start with <code>$</code>):<br>
+     *           Returns an array of object keys for the object value matching the path. If multiple
+     *           paths are matched, returns the length of the first matching object. If <code>path
+     *           </code> doesn't exist or the value at <code>path</code> is not an array, an error is
+     *           raised.
+     *     </ul>
+     *     If <code>key</code> doesn't exist, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\": 1.0, \"b\": {\"a\": {\"x\": 1, \"y\": 2}, \"b\": 2.5, \"c\": true}}").get();
+     * var res = Json.objkeys(client, "doc", ".").get();
+     * assert Arrays.equals((Object[]) res, new Object[] { "a", "b" }); // key names for the object matching the path `.` as it is the only match.
+     * res = Json.objkeys(client, "doc", "$.b").get();
+     * assert Arrays.equals((Object[]) res, new Object[][] { { "a", "b", "c" } }); // key names as a nested list for objects matching the JSONPath `$.b`.
+     * }</pre>
+     */
+    public static CompletableFuture<Object[]> objkeys(
```

Comment:
objkeys returns an array is all cases?  well at least _one_ of the commands acts decently.

Reply:
`Object[][]` for JSONPath and `Object[]` for legacy path. So common denominator is `Object[]`.